sdk/client-core/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -154,13 +156,32 @@ public ResponseBodyMode getResponseBodyMode() {
     }
 
     /**
-     * Adds a header to the HTTP request.
-     * <p>
-     * If a header with the given name exists the {@code value} is added to the existing header (comma-separated),
-     * otherwise a new header is created.
+     * Whether the response headers should be eagerly converted.
+     *
+     * @return Whether the response headers should be eagerly converted.
+     */
+    public boolean isEagerlyConvertHeaders() {
```

Comment:
Now that this is a public API we should have a more user-friendly name for this method. What conversion do we do here? Is it eagerly deserializing headers into models -  `isEagerlyDeserializeHeaders()`?

Reply:
This is used to determine whether the HTTP header representation returned by the HttpClient implementation is eagerly converted into HttpHeaders in client-core. This is something we can remove for now and always use lazy conversion, maybe with some simple metrics on when to perform a full conversion.

Some back story to explain this. There are a few high performance Azure Core-based SDKs which use deserialized headers. When we had our last major round of performance optimizations we found that a healthy portion of CPU time was spent dealing with looking performance behaviors of underlying HTTP header represenations. For example, OkHttp uses an array that alternates key, value to represent headers, getting all values of a single header name required iterating the entire array, so you can see where this ends up if you're doing a lot of header look ups. So, this runtime behavior was added to eagerly convert to the HttpHeaders in a way that performed a single iteration over the OkHttp headers.

This all can be replaced with a mechanism where the headers are either converted as they're accessed or after a certain number of accesses. Especially if we don't continue the concept of deserialized header objects in client-core based libraries.

Reply:
So, I think we can always use lazy conversion and remove this from RequestOptions. For an end customer, this doesn't have to be configurable.

Reply:
When creating an `OkHttpResponse`, we either convert all the OkHttp headers to our own `HttpHeaders` type or do so lazily on a one-by-one basis when a user tries to access a value. In Azure Core we do the same for Netty headers.

Reply:
Created #39657 to track this.